theano/gpuarray/pool.py

```diff
@@ -411,15 +412,102 @@ def make_node(self, inp, eval_point, ws, stride=None, pad=None):
 
         return Apply(self, [inp, eval_point, ws, stride, pad], [eval_point.type()])
 
+
+class GpuRoIPoolOp(CGpuKernelBase):
+
+    __props__ = ('spatial_scale', 'pooled_h', 'pooled_w')
+    _f16_ok = True
```

Comment:
The code currently don't support float16 storage. So remove that. idem for the second class.

For now, put float16 aside. First make it work in float32/64.